doc/user/content/overview/architecture.md

```diff
@@ -8,23 +8,124 @@ menu:
+
+## Layers
+
+Now we're ready to dive deeper into each of the three layers. Each layer has its own SQL nouns and concepts. This section will overview each layer, what it does, and how all of the nouns and concepts work.
 
 ### Storage
 
+Storage is responsible for ensuring that data read from external systems (e.g. a Kafka topic), are granted a timestamp and then durably persisted to storage (currently S3). Storage is also responsible for writing data from Materialize back to external systems (e.g. Kafka again). Storage handles connecting to/authenticating these external systems and converting data from/to external formats (e.g. Protobuf) to/from the Materialize internal binary representation of data.
+
+The key concepts and nouns in storage that users have to interact with are `CONNECTION`s, `SOURCE`s, and `SINK`s.
+
+#### Connections
+
+A connection describes how to connect and authenticate to an external system you want Materialize to read data from/to. A connection is a logical entity in Materialize in that it exists purely to make subsequent usage of Materialize easier for the user, and creating a connection itself doesn't cause anything to happen.
+
+#### Sources
+
+Sources describe external systems you want Materialize to read data from, and provide details about how to decode and interpret that data. Importantly, and unlike connections, as soon as you create a source, Materialize starts reading from the specified external system, and decoding, interpreting and ultimately writing the data out to persistent storage. The amount of computational resources (CPU and memory) available to handle all of the data from a source can be scaled up and down by changing the source's size with `ALTER SOURCE`.
+
+#### Sinks
+
+Sinks are the inverse of sources and represent a connection to an external stream where Materialize outputs data from its persistent storage. Sinks don't read directly from the compute layer [todo: is this true?]. Sinks, like sources begin their works immediately. When a user defines a sink over a materialized view, source, or table, Materialize automatically generates the required schema and writes down the stream of changes to that materialized view, source, or table. The amount of computational resources (CPU and memory) available to encode all of the data that needs to be written can be scaled up and down by changing the sink's size with `ALTER SINK`.
+
 ### Compute
 
-### Adapter
+Compute is responsible for reading in input collections from `STORAGE` (by reading from S3 instead of the upstream external system) [todo: is this too prescriptive? I want to get across that compute doesn't touch kafka], and transforming those collections into ouput collections that contain correct answers for the requested computations.
```

Comment:
The definition for the compute layer [here]() is more straightforward:

> The Compute layer is tasked with efficiently computing and maintaining views over explicitly timestamped histories of data collections.

Mentioning that we ingest data once for each source and work out of storage should maybe go under Storage?